### ISSUE TITLE
8卡 A100-40G 可以做34B模型sft吗？

### ISSUE THREAD

**Issue by Modberge** [01-ai/Yi#478]
### Reminder

- [X] I have searched the Github Discussion and issues and have not found anything similar to this.

### Environment

```markdown
- OS:Ubuntu 20.04
- Python:3.10
- PyTorch:2.0.1+cu118
- CUDA:11.8
```

### Current Behavior

8卡 A100-40G 可以做34B模型sft吗？内存 2T，如果可以的话 应该怎么改运行脚本？
目前的脚本设置：

#/usr/bin/env bash

cd "$(dirname "${BASH_SOURCE[0]}")/../sft/"

export CUDA_VISIBLE_DEVICES=0,1,2,3 #limit parallelism to avoid cpu oom

deepspeed --master_port 6778 main.py \
	--data_path ../yi_example_dataset/ \
	--model_name_or_path /mnt/bn/Yi-34B  \
	--per_device_train_batch_size 1 \
	--per_device_eval_batch_size 1 \
	--max_seq_len 8 \
	--learning_rate 2e-6 \
	--weight_decay 0. \
	--num_train_epochs 4 \
	--training_debug_steps 20 \
	--gradient_accumulation_steps 1 \
	--lr_scheduler_type cosine \
	--num_warmup_steps 0 \
	--seed 1234 \
	--gradient_checkpointing \
	--zero_stage 2 \
	--deepspeed \
	--offload \
	--output_dir /mnt/bn/Yi_model/sft_model/Yi_34B_SFT_v1

### Steps to Reproduce

bash run.sh

**Comment by Yimi81** [comment]
可以做34B非全量的微调，目前官方的sft代码没在维护了，而且有很多现有的微调库支持各种方法对Yi模型的微调，如[llama_factory](https://github.com/hiyouga/LLaMA-Factory), [swift](https://github.com/modelscope/swift), [firefly](https://github.com/yangjianxin1/Firefly)等. 请选择一个然后进行SFT微调
